src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -512,6 +517,9 @@ public override bool WorksOnAnyCollection(HashSet<string> collections)
             if (_referencedCollections == null)
                 return false;
 
+            if (HandleAllDocs)
```

Comment:
Couldn't we have this check in `base.WorksOnAnyCollection()`?

Reply:
Moved.
Found an issue that was fixed here: https://github.com/ravendb/ravendb/pull/20094.